Library/Homebrew/formula.rb

```diff
@@ -934,9 +934,14 @@ def exec_cmd(cmd, args, out, logfn)
 
   def stage
     active_spec.stage do
+      old_home = ENV["HOME"]
       @buildpath = Pathname.pwd
+      envhome = buildpath/".brew_home"
+      mkdir_p envhome
+      ENV["HOME"] = envhome
       yield
       @buildpath = nil
+      ENV["HOME"] = old_home
```

Comment:
Since we're mutating global state, we need to use a proper cleanup idiom here:

```rb
def stage
  active_spec.stage do
    @buildpath = Pathname.pwd
    env_home = buildpath/".brew_home"
    mkdir_p env_home

    old_home, ENV["HOME"] = ENV["HOME"], env_home

    begin
      yield
    ensure
      @buildpath = nil
      ENV["HOME"] = old_home
    end
  end
end
```